### PR TITLE
remove more (all?) personas cruft

### LINKED ISSUE #12030: remove references to ADDON_PERSONA
And other persona related code.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #12030

I (attempted) to make the API backwards compatible by keeping references to `persona` in the constants.  We probably want to tidy that up too at some point, but in a v3 compatible way (and `ADDON_THEME` while we're at it)

### CODE REVIEW

File: src/olympia/addons/indexers.py [hunk 1]
```diff
@@ -172,17 +172,6 @@ def get_mapping(cls):
...
-                        'type': 'object',
-                        'properties': {
-                            'accentcolor': {'type': 'keyword', 'index': False},
-                            'author': {'type': 'keyword', 'index': False},
-                            'header': {'type': 'keyword', 'index': False},
-                            'footer': {'type': 'keyword', 'index': False},
-                            'is_new': {'type': 'boolean', 'index': False},
-                            'textcolor': {'type': 'keyword', 'index': False},
-                        }
-                    },
```
Comment (EnTeQuAk) [inline src/olympia/addons/indexers.py]: This requires a reindex.

File: src/olympia/constants/base.py [hunk 2]
```diff
@@ -113,13 +96,13 @@
 ADDON_LPADDON = 6
 ADDON_PLUGIN = 7
 ADDON_API = 8  # not actually a type but used to identify extensions + themes
-ADDON_PERSONA = 9
+_ADDON__PERSONA = 9  # Deprecated - should only exist as STATUS_DELETED
```
Comment (EnTeQuAk) [inline src/olympia/constants/base.py]: why the double underscore between `addon` and `persona`?

Comment (eviljeff) [reply src/olympia/constants/base.py]: oh, it was just easier to see what usage I still needed to remove/change. I can change it to `_ADDON_PERSONA`.

Comment (eviljeff) [reply src/olympia/constants/base.py]: https://github.com/mozilla/addons-server/pull/12031/commits/7ff42233c49c2f18a61900e3466ddb3dacda0730

File: src/olympia/addons/indexers.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/addons/indexers.py]: started on dev, for now

**Pull request merged**